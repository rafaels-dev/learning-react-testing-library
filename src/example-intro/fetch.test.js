import React from 'react';
import { render, fireEvent, waitForElement, wait } from '@testing-library/react';

import axiosMock from 'axios';

import Fetch from './fetch';

test.only('loads and displays greetings', async () => {
    const url = '/greeting';

    axiosMock.get.mockResolvedValueOnce({
        data: { greeting: 'hello there' }
    });

    const { getByText, getByRole, container, asFragment } = render(<Fetch url={url} />);
    fireEvent.click(getByText('Load Greeting'));
    const greetingTextNode = await waitForElement(
        () => getByRole('heading')
    );

    expect(axiosMock.get).toHaveBeenCalledTimes(1);
    expect(axiosMock.get).toHaveBeenCalledWith(url);
    expect(getByRole('heading')).toHaveTextContent('hello there');
    expect(getByRole('button')).toHaveAttribute('disabled');
    expect(container).toMatchSnapshot();
});