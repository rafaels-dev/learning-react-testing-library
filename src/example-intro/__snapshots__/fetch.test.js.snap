// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greetings 1`] = `
<div>
  <div>
    <button
      disabled=""
    >
      Ok
    </button>
    <h1>
      hello there
    </h1>
  </div>
</div>
`;
